### PR TITLE
File processing change + Docker support

### PR DESCRIPTION
**Pull request by MatthewDeSouza**
This pull request includes significant updates to the Docker deployment, configuration handling, and error logging for the `artisan-stock-checker` project. The most important changes involve the introduction of Docker support, improvements to configuration management, and enhancements to error logging.

### Docker Deployment:
* Added Docker support with a `Dockerfile` and a `docker-compose` configuration to streamline the deployment process. (`Dockerfile`, `compose.yaml`, `.env.example`) [[1]](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R1-R13) [[2]](diffhunk://#diff-facaded51b7d1c9656ae43b449b69aa398fee9129321a0001d97048c00c8cc1cR1-R10) [[3]](diffhunk://#diff-a3046da0d15a27e89f2afe639b25748a7ad4d9290af3e7b1b6c1a5533c8f0a8cR1-R10)

### Configuration Handling:
* Updated configuration handling to use environment variables and default values, making it more flexible and easier to manage. (`discord_webhook_script/check_artisan_stock_webhook.py`, `discord_webhook_script/config_handler.py`, `basic_script/config_handler.py`) [[1]](diffhunk://#diff-2c6a00316039ba946e8e67d4e3f56f079d7a1fc53d1ed24bb46e2e7e59ec3401L15-R38) [[2]](diffhunk://#diff-5d22307e91f9e9b696e591d6d0a394e1765bdcc95d3d1b5bc0e9e7dd78f0eccfR7-R29) [[3]](diffhunk://#diff-5d74358f78942c2fec7a09ddb1fe8862de5b377df66fb592c2b4453e653cffc5R7-R28)

### Error Logging:
* Enhanced error logging by ensuring logs are stored in a specific directory defined by an environment variable, improving log management and troubleshooting. (`discord_webhook_script/error_logger.py`)

### Documentation:
* Added a link to Docker deployment documentation in the `README.md` to guide users through the deployment process. (`README.md`)

### CI/CD Integration:
* Implemented a GitHub Actions workflow to automate the building and pushing of Docker images to DockerHub, ensuring continuous integration and delivery. (`.github/workflows/docker_image.yml`)

### CODE REVIEW

File: discord_webhook_script/check_artisan_stock_webhook.py [hunk 1]
```diff
@@ -12,18 +13,29 @@
 
 utc_time = datetime.now(timezone.utc).strftime("%Y-%m-%d_%H-%M-%S")
 
-stock_delay = config_handler.read("config.cfg", "stock", "stock_delay")
-cart_delay = config_handler.read("config.cfg", "stock", "cart_delay")
-batch_delay = config_handler.read("config.cfg", "stock", "batch_delay")
+# Introduce CONFIG_PATH variable, get from environment variable if set
+CONFIG_PATH = os.environ.get('CONFIG_PATH', '.')
```
Comment (Kuuuube) [inline discord_webhook_script/check_artisan_stock_webhook.py]: `CONFIG_PATH` is way too generic it needs to be something like `ARTISAN_STOCK_CHECKER_CONFIG_DIR` to ensure it won't get stepped on or be stepping on an env var something else would use. And i think "PATH" makes it sound like this is the path to the config file specifically not the directory for storing any working files.

File: discord_webhook_script/webhook_handler.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def roles_dict(model, hardness, config_file=DEFAULT_CONFIG_FILE):
                     config_file, "webhook_role_pings", "role_CS_Raiden"
                 ),
             }
-
+            model = model_full;
```
Comment (Kuuuube) [inline discord_webhook_script/webhook_handler.py]: Either name it `model_full` and `return roles_dict[model_full]` without touching `model` or leave this code alone. This change is logically pointless and just makes it more confusing.

**Review by Kuuuube — DISMISSED** [review]
Looks good just some minor things.

**Comment by MatthewDeSouza** [comment]
Staged the wrong file apparently in the first commit, all suggestions implemented.

**Review by Kuuuube — APPROVED** [review]

**Pull request merged**